bin/conform-templates.php

```diff
@@ -0,0 +1,299 @@
+    $content = file_get_contents('LICENSE.md');
+    if (preg_match('/Copyright \(c\) (\d{4})/', $content, $m)) {
+        $year = $m[1];
+    } else {
+        fwrite(STDERR, 'Cannot match year in current LICENSE.md file. Using current year only.' . PHP_EOL);
+    }
+}
+
+$content = file_get_contents(__DIR__ . '/../template/LICENSE.md');
+$content = str_replace(
+    '{year}',
+    $year < $currentYear ? $year . '-' . $currentYear : $currentYear,
+    $content
+);
+
+file_put_contents('LICENSE.md', $content);
+
+// .coveralls.yml
+file_put_contents('.coveralls.yml', file_get_contents(__DIR__ . '/../template/.coveralls.yml'));
+
+// check if repository has documentation
+$hasDocs = file_exists('mkdocs.yml');
+
+// .gitattributes
+$content = file_get_contents(__DIR__ . '/../template/.gitattributes');
+if (! $hasDocs) {
+    $content = preg_replace('/\/mkdocs\.yml export-ignore\s+/', '', $content);
+}
+if (file_exists('phpbench.json')) {
+    if (file_exists('benchmark') && is_dir('benchmark')) {
+        $content = '/benchmark export-ignore' . PHP_EOL . $content;
+    }
+    if (file_exists('benchmarks') && is_dir('benchmarks')) {
+        $content = '/benchmarks export-ignore' . PHP_EOL . $content;
+    }
+    $content = preg_replace(
+        '/\/phpcs\.xml export-ignore/',
+        '/phpbench.json export-ignore' . PHP_EOL . '/phpcs.xml export-ignore',
+        $content
+    );
+}
+if (file_exists('.docheader')) {
+    if (isset($composer['require-dev']['malukenho/docheader'])) {
+        $content = preg_replace(
+            '/\/\.coveralls.yml export-ignore/',
+            '/.coveralls.yml export-ignore' . PHP_EOL . '/.docheader export-ignore',
+            $content
+        );
+    } else {
+        unlink('.docheader');
+    }
+}
+file_put_contents('.gitattributes', $content);
+
+// .gitignore
+$content = file_get_contents(__DIR__ . '/../template/.gitignore');
+if (! $hasDocs) {
+    $content = preg_replace('/^docs\/html\/\s+/m', '', $content);
+    $content = preg_replace('/^zf-mkdoc-theme\/\s+/m', '', $content);
+    $content = preg_replace('/^zf-mkdoc-theme\.tgz\s+/m', '', $content);
+}
+file_put_contents('.gitignore', $content);
+
+// .travis.yml - create only when does not exist
+if (! file_exists('.travis.yml')) {
+    copy(__DIR__ . '/../template/.travis.yml', '.travis.yml');
+}
+
+// composer.json
+// - checks repository description
+// - checks order of sections
+// - removes default type library
+// - updates scripts
+// - updates license
+// - updates support links
+// - updates "config"
+// - removes "minimum-stability" and "prefer-stable"
+// - checks keywords
+// @todo: check branch-alias
+
+$templateContent = json_decode(file_get_contents(__DIR__ . '/../template/composer.json'), true);
+
+$sectionOrder = [];
+foreach ($templateContent as $section => $value) {
+    $sectionOrder[] = $section;
+}
+
+$content = $composer;
+if (isset($content['type']) && $content['type'] === 'library') {
+    unset($content['type']);
```

Comment:
Why are you unsetting this? Shouldn't this stay?

Reply:
I'm unsetting only in case of "type=library" as it is default type.
See https://getcomposer.org/doc/04-schema.md#type
And https://github.com/zendframework/zend-stdlib/pull/85#pullrequestreview-71986270

Reply:
Okay; wasn't sure about that as I recently saw PRs that added the `"type": "library"` entry.